modules/cudaarithm/src/cuda/absdiff_scalar.cu

```diff
@@ -78,33 +78,57 @@ namespace
     };
 
     template <typename SrcType, typename ScalarDepth>
-    void absDiffScalarImpl(const GpuMat& src, double value, GpuMat& dst, Stream& stream)
+    void absDiffScalarImpl(const GpuMat& src, cv::Scalar value, GpuMat& dst, Stream& stream)
```

Comment:
Does `cv::Scalar value` => `const cv::Scalar& value` work?
What is about "double value[4]"?